### PR TITLE
add new sdk samples

### PR DESCRIPTION
**Pull request by qubitron**
# Description

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure-Samples/azureai-samples/blob/main/CONTRIBUTING.md)
- [x] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [x] This notebook or file is added to the [CODEOWNERS](https://github.com/Azure-Samples/azureai-samples/blob/main/.github/CODEOWNERS) file, pointing to the author or the author's team.

### CODE REVIEW

**Comment by qubitron** [comment]
@kdestin I've added unit tests and fixed a bunch of linter errors -- I suppressed errors around module imports because these files are structured for docs, and lack of type hints because I'm not sure if we want type hints in our docs to keep them simple (but I can easily add those if needed)

File: scenarios/rag/custom-rag-app/packages/azure_ai_inference-1.0.0b5-py3-none-any.whl
Comment (kdestin) [inline scenarios/rag/custom-rag-app/packages/azure_ai_inference-1.0.0b5-py3-none-any.whl:1]: Can we install this from the public release?

File: scenarios/rag/custom-rag-app/packages/azure_ai_projects-1.0.0b1-py3-none-any.whl
Comment (kdestin) [inline scenarios/rag/custom-rag-app/packages/azure_ai_projects-1.0.0b1-py3-none-any.whl:1]: Can we install this from the public release?

File: scenarios/inference/chat-app/chat-simple.py
Comment (kdestin) [inline scenarios/inference/chat-app/chat-simple.py:1]: Can we add a README.md for this sample?

File: scenarios/rag/custom-rag-app/README.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+Check out the following files for implementation details:
+ - [create_search_index.py](): creates an Azure AI search index 
+    - Uses data in [assets/products.csv]() to generate the index
+ - [get_product_documents.py](): implements document retreival based on user queries
+    - Using [assets/intent_mapping.prompty]() for generating a search query from the conversation
+ - [chat_with_products.py](): implements chat that grounds response in retrieved documents
+    - Uses [assets/grounded_chat.prompty]() for formatting retrieved documents into the chat prompt
+    - Implements the [Microsoft AI Chat protocol](https://github.com/microsoft/ai-chat-protocol) for compatibility with evaluation and front-ends
+
```
Comment (kdestin) [inline scenarios/rag/custom-rag-app/README.md:26]: Could be worth adding in a link to relevant documentation near the start of this sample, similar to what's in the jupyter template.

image

**Pull request closed without merging**